src/Core/ConstraintValidator/TypedRegexValidator.php

```diff
@@ -41,6 +41,10 @@
  */
 class TypedRegexValidator extends ConstraintValidator
 {
+    const GENERIC_NAME_CHARS = '<>={}';
+    const MESSAGE_CHARS = '<>{}';
+    const NAME_CHARS = '0-9!<>,;?=+()@#"�{}_$%:';
```

Comment:
I think placeholders in this class are served by `getPattern()` function. See `TypedRegexValidator::getPattern()`

Reply:
Indeed it seems they do. Problem is that its a private function. Would it make sense to make it public and add any non-existing patterns there?

Reply:
Yes it makes sense to make it public. It could then be used by external classes to fetch a pattern they need 👍

Reply:
I investigated this further and there are problems with using getPattern function. For example TYPE_GENERIC_NAME returns pattern of /^[^<>={}]*$/u of course i only need <>={} part for description. 

Now I see three ways of solving this:

1. Create a function getPatternDescription which would remove all the regex stuff and leave pattern chars only. It would be pretty convenient but I couldn't find a nice way to do it, writing a big regex to consider all possible cases could make it pretty error-prone. 
2. Use the same function but hardcode those patterns instead I.E 
```
$typePatterns = [
       TypedRegex::TYPE_GENERIC_NAME => '<>={}'
  ]
  if (isset($typePatterns[$type])) {
       return $typePatterns[$type];
  }
  throw new InvalidArgumentException();
]
```

 Main disadvantage here is that if I need a pattern I need to have TypedRegexValidator as dependancy

3. Leave it the way it's done above the comment. Disadvantage is that constants are not that as intuitive as having a function based on the constants like TypedRegex::TYPE_GENERIC_NAME (which are already used when adding regex constraint), however we also don't need extra dependency if we want to get patterns. 

I think 1 would be the best way to do it if there was an easy way to remove all the regex stuff and leave only a pattern and getPatternDescription would be very simple `return $this->cleanRegex($this->getPattern($type)) `. If not I would go with 3 as it avoids an unnecessary dependency.

Reply:
Let's go for 3. because we are not needing regexps here.

In future, it would be nice to implement 1. and make it a static function (and maybe make `getPattern($type)` static too because we dont need it to be an instance so, as you said, it's a shame to need a dependency injection just for that.